features/enlinkd/adapters/common/src/main/java/org/opennms/netmgt/enlinkd/common/TopologyUpdater.java

```diff
@@ -217,8 +229,10 @@ private static IpInterfaceTopologyEntity getPrimary(IpInterfaceTopologyEntity n1
     public abstract OnmsTopology buildTopology();
 
     @Override
-    public synchronized OnmsTopology getTopology() {
-        return m_topology.clone();
+    public OnmsTopology getTopology() {
+        synchronized(m_topology) {
```

Comment:
We may want to add a synchronized block in the setter too.

Reply:
Yep, good point :)